packages/repository/src/entity/proto.test.ts

```diff
@@ -3,84 +3,59 @@ import { DeepReadonly } from "./types"
 import { TestEntityData, describe, expect, expectTypeOf, it } from "vitest"
 
 describe("proto", () => {
-  it("Keep type of given input data", ({ syncKeys }) => {
-    const context = createInternalEntity<TestEntityData>(syncKeys)
-
-    const entity = new context({
-      id: "1",
-      foo: "foo",
-      bar: 1,
-    })
-
-    const result = entity.toObject()
-    expect(result).toEqual({
-      foo: "foo",
-      bar: 1,
-      id: "1",
-    })
-    expectTypeOf(result).toEqualTypeOf<
-      DeepReadonly<{ foo: string; bar: number; id: string }>
-    >()
-  })
-
   it("Update method returns new instance with extended data", ({
     syncKeys,
+    fakeData,
   }) => {
     const context = createInternalEntity<TestEntityData>(syncKeys)
 
-    const entity = new context({
-      id: "1",
-      foo: "foo",
-      bar: 1,
-    })
+    const entity = new context({ ...fakeData, id: "1" })
```

Comment:
Ta sekcja mi rozjechała banie 😄

Funkcja createInternalEntity to funkcja zwracająca deklaracje klasy - dlaczego tak jest?
Jakie mamy plusy związane przez takei rozwiązanie? 📖

Reply:
Hehe, no tego można było się spodziewać to niecodzienny manerw, ale jak się nad tym zastanowić to nie ma w tym nic dziwnego że tak można i że się da itd. :smile: W końcu funkcję są first-class citizens w JS.

Dlaczego tak jest? 
Generalnie klasa wewnętrzna jest dedykowana do każdego typu encji - dlatego to fabryka przyjmuje generyk z modelem (`TSchema`) i punkty do synchronizacji (`syncKeys`) - bo one są niezmienne w obrębie zdefiniowanego modelu encji. Rzeczy w samej klasie się wymieniają - bo to puszka na te wymienne rzeczy w encji - czyli przede wszystkim dane, które mrozimy w konstruktorze oraz resetujemy stan synchronizacji (nowa instancja `SyncMap` - ale na niezmiennym zestawie kluczy).

Więc w takim podejściu kod odpowiada modelowi i podziałowi zadań i wydaje mi się że łatwiej go przetestować ;)